app/views/statistics/_collaborators.html.slim

```diff
@@ -6,7 +6,8 @@
 
 h2 Collaborators
 
-= link_to "Mailing List Export", :controller => 'statistics', :action => 'export_csv'
+- if user_signed_in?
```

Comment:
Only the current collection owner (and sub owners) should be able to see and download the spreadsheet.